### ISSUE TITLE
Tooltip2 opens when dialog closes.

### ISSUE THREAD

**Issue by dlech** [palantir/blueprint#5169]
#### Environment

- __Package version(s)__: @blueprintjs/popover2 v1.0.0-rc.0
- __Operating System__: Ubuntu 20.04
- __Browser name and version__: Firefox 98.0

#### Code Sandbox

Link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/blueprint-sandbox-et9xy)): https://codesandbox.io/s/blueprint-sandbox-forked-wdutny?file=/src/index.tsx

#### Steps to reproduce

1. Click the test button, dialog will open
1. Close the dialog, the mouse cursor is no longer over the button
1. The tooltip opens

#### Actual behavior

The tooltip opens when the mouse is not over the button.

https://user-images.githubusercontent.com/963645/158651875-e49e775e-306c-4e90-ab2e-96a0cdc21ade.mp4

#### Expected behavior

The tooltip should not open.

#### Possible solution

Since the dialog takes focus from the button and then gives it back when the dialog closes, it is triggering the tooltip. There is some logic already (`lostFocusOnSamePage`) to try to detect when focus is triggered due to leaving the page and coming back to avoid opening the tooltip. It could be possible to extend this logic to also detect the case focus coming from a dialog closing.

**Comment by adidahiya** [comment]
Try setting `shouldReturnFocusOnClose={false}` on the Dialog: https://blueprintjs.com/docs/versions/4/#core/components/dialog

**Comment by dlech** [comment]
Yes, I figured out this workaround already. It probably isn't the best solution for accessibility though.
